### PR TITLE
Fix mul overload

### PR DESCRIPTION
**Pull request by wsmoses**

### CODE REVIEW

File: src/Overlay.jl [hunk 1]
```diff
@@ -130,14 +130,18 @@ for (cT, aT, bT) in (
             if any(Base.Fix2(isa, TracedRArray) ∘ ancestor, (C, A, B))
                 TracedLinearAlgebra.overloaded_mul!(C, A, B, α, β)
             else
-                LinearAlgebra._mul!(C, A, B, α, β)
+                LinearAlgebra.mul!(C, A, B, α, β)
```
Comment (avik-pal) [inline src/Overlay.jl:133]: this will very likely trigger the illegal instruction https://github.com/EnzymeAD/Reactant.jl/issues/393

**Comment by wsmoses** [comment]
@avik-pal it looks like this is hitting some scalar index stuff?

**Comment by wsmoses** [comment]
Yeah we need to change aos to anytracedarray

File: src/Overlay.jl (hunk 1 above)
Comment (wsmoses) [reply src/Overlay.jl:134]: There was an absint behavior that we had a while ago but was accidentally removed in a refactor which I re added with the last cuda stuff.

I hope it means we won't hit it

Comment (avik-pal) [reply src/Overlay.jl:134]: I checked it again in the stack PR, the crash persists

Comment (wsmoses) [reply src/Overlay.jl:134]: Well darn, okay yeah if CI yells I think we need to make a separate noinline function which is called in the override, which itself calls mul

Comment (avik-pal) [reply src/Overlay.jl:134]: lets add a test like https://github.com/EnzymeAD/Reactant.jl/blob/main/test/basic.jl#L864 to verify this

**Comment by wsmoses** [comment]
Okay @avik-pal tesfs pass so I’m going to merge

**Pull request merged**